MathLib/LinAlg/RowColumnIndices.h

```diff
@@ -14,19 +14,25 @@
 
 namespace MathLib
 {
+// A dummy vector as the default value of the third argument of
+// the constructor.
+static std::vector<std::size_t> dummy_non_ghost_local_ids;
 
 template <typename IDX_TYPE>
 struct RowColumnIndices
 {
 	typedef typename std::vector<IDX_TYPE> LineIndex;
-	RowColumnIndices(LineIndex const& rows_, LineIndex const& columns_)
-		: rows(rows_), columns(columns_)
+	RowColumnIndices(LineIndex const& rows_, LineIndex const& columns_,
+	                 const std::vector<std::size_t>& non_ghost_local_ids_
+	                 = dummy_non_ghost_local_ids)
+		: rows(rows_), columns(columns_), non_ghost_local_ids(non_ghost_local_ids_)
 	{ }
 
 	LineIndex const& rows;
 	LineIndex const& columns;
-};
 
+	std::vector<std::size_t> const& non_ghost_local_ids;
```

Comment:
Is it possible to avoid the non_ghost_local_ids here and move it entirely to AssemblerLib?
I'm asking, because this member is used outside of AssemblerLib only once in the GroundwaterFlowFEM, which should become independent of any mpi related code.